clang/test/Analysis/engine/expr-engine-cxx-crash.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clang_analyze_cc1 -analyzer-checker=core.DivideZero -std=c++23 -verify %s
+// expected-no-diagnostics
+
+struct S
+{
+    constexpr auto operator==(this auto, S)
+    {
+        return true;
+    }
+};
+
+int main()
+{
+    return S {} == S {};
+}
```

Comment:
```suggestion
struct S {
  bool operator==(this auto, S) {
    return true;
  }
};

int use_deducing_this() {
  return S{} == S{};
}
```

Reply:
Speaking of this test, it might be better to dedicate a file for "deducing this".

Reply:
fixed in 5ac52a8